docs/Developer-Guide.md

```diff
@@ -502,6 +490,8 @@ bash-4.2# exit
 exit
 ```
 
+`kata-runtime exec` has a command-line option `runtime-namespace`, which is used to specify under which namespace the containers(Pods) are created. By default, it is set to `k8s.io` and works for containerd, for other upper runtimes, you may need to set the namespace by `runtime-namespace` option.
+
```

Comment:
It's unclear to me what "namespace" means in this context.  It's not a k8s namespace, and I don't think it's a kernel namespace, so what?

Reply:
It’s more of a concept that was introduced in containerd to help facilitate working in multi tenant environments. The containers/pods/images that are running are “namespaced”, somewhat similar to what is done in kube itself. Check the markdown link in Bin’s latest commit. I’m happy to see it further clarified if necessary, but i hope the link helps.

Reply:
Ok.. but if it's a containerd concept, what does it mean to Kata?  I really think we need to describe this in term's of *Kata*'s behaviour, not the behaviour of a different package that might or might not be used.

Reply:
@dgibson  - I see your point; what is the address of the UDS for the shim-monitor when integrated with CRI-O?

Ultimately, depending on how we're integrated in the stack, we'll need to provide a way to specify the unique UDS path associated with a container. In the common case, this is 
```containerd-shim/k8s.io/sandbox.id/shim-monitor.sock``` 

Depending on how the stack is configured, though, the socket could be at s/k8s.io/default or s/k8s.io/sweeeeeeet.

Reply:
I'm assuming UDS here means "Unix Domain Socket"?  I don't know how that works with CRI-O, my experience is all at lower levels of the stack.

It sounds like the parameter should really be giving an explicit path (or at least directory) for the control socket.  How the next layer up arranges the control sockets by namespace is its business, not Kata's.

Reply:
Hi @dgibson , the address is invisible from outside of shim process, the path is [fixed](https://github.com/kata-containers/kata-containers/blob/main/src/runtime/containerd-shim-v2/shim_management.go#L196):

```
filepath.Join(string(filepath.Separator), "containerd-shim", ns, id, "shim-monitor.sock")
```

The namespace(`ns`) is passed from upper runtime and untouched in shim.